src/beta_inc.jl

```diff
@@ -0,0 +1,884 @@
+            mu = k
+        end
+        t = mu
+        d = exp(-t)
+    end
+
+    ans = brcmp1(mu,a,b,x,y,true)/a
+    if n == 1 || ans == 0.0
+        return ans
+    end
+    nm1 = n -1
+    w = d
+
+    k = 0
+    if b <= 1.0
+        @goto l40
+    elseif y > 1.0e-4
+        @goto l20
+    else
+        k = nm1
+        @goto l30
+    end
+
+    @label l20
+     r = (b - 1.0)*x/y - a
+     if r < 1.0
+        @goto l40
+     end
+     k = t = nm1
+     if r < t
+        k = r
+     end
+    # ADD INC TERMS OF SERIES
+    @label l30
+     for i = 1:k
+        l = i -1
+        d *= ((apb + l)/(ap1 + l))*x
+        w += d
+     end
+     if k == nm1
+        return ans*w
+     end
+    # ADD REMAINING TERMS
+    @label l40
+     kp1 = k + 1
+     for i = kp1:nm1
+        l = i - 1
+        d *= ((apb + l)/(ap1 + l))*x
+        w += d
+        if d <= epps*w
+            break
+        end
+    end
+    return ans*w
+end
+#SIGNIFICANT DIGIT COMPUTATION OF INCOMPLETE BETA FUNCTION RATIOS
+#by ARMIDO R. DIDONATO AND ALFRED H. MORRIS, JR.
+#ACM Transactions on Mathematical Software. Vol18, No3, September1992, Pages360-373
+
+"""
+    beta_inc(a,b,x,y)
+
+Computes Incomplete Beta Function Ratios given by:
+```math
+I_{x}(a,b) = G(a,b) \\int_{0}^{x} t^{a-1}(1-t)^{b-1} dt,
+```
+and ``I_{y}(a,b) = 1.0 - I_{x}(a,b)``.
+given
+``B(a,b) = 1/G(a,b) = \\Gamma(a)\\Gamma(b)/\\Gamma(a+b)`` and y = 1.0 - x
+"""
+function beta_inc(a::Float64, b::Float64, x::Float64, y::Float64)
+    ans_x = 0.0
+    ans_y = 0.0
+    lambda = a - (a+b)*x
+    # if a < 0.0 || b < 0.0
+    #     println("1")
+    #     return (0.0,0.0)
+    # elseif a == 0.0 && b == 0.0
+    #     println("2")
+    #     return (0.0,0.0)
+    # elseif x < 0.0 || x > 1.0
+    #     println("3")
+    #     return (0.0,0.0)
+    # elseif y < 0.0 || y > 1.0
+    #     println("4")
+    #     return (0.0,0.0)
+    # else
+    #     z = x + y - 1.0
+    #     if abs(z) > 3.0*eps()
+    #         println("5")
+    #         return (0.0,0.0)         # ERROR HANDLING
+    #     end
+    # end
+
+    if x == 0.0
+        return (0.0,1.0)
+    elseif y == 0.0
+        return (1.0,0.0)
+    elseif a == 0.0
+        return (1.0,0.0)
+    elseif b == 0.0
+        return (0.0,1.0)
+    end
+
+    epps = max(eps(), 1.0e-15)
+    if max(a,b) < 1.0E-3 * epps
+        return (b/(a+b), a/(a+b))
+    end
+    ind = 0.0
+    a0 = a
+    b0 = b
+    x0 = x
+    y0 = y
+    if min(a0,b0) > 1.0
+        @goto l30
+    end
+#PROCEDURE FOR A0<=1 OR B0<=1
+    if x <= 0.5
+        @goto l10
+    end
+    ind = 1.0
+    a0 = b
+    b0 = a
+    y0 = x
+    x0 = y
+
+    @label l10
+     if b0 < min(epps, epps*a0)
+        @goto l80
+     elseif a0 < min(epps, epps*b0)
+        @goto l90
+     elseif max(a0,b0) > 1.0
+        @goto l20
+     elseif a0 >= min(0.2, b0) || x0^a0 <= 0.9
+        @goto l100
+     elseif x0 >= 0.3
+        @goto l110
+     else
+        n = 20
+        @goto l130
+     end
+
+    @label l20
```

Comment:
since the above block doesn't flow through, here to `@label l30` can be moved to `@goto l20`